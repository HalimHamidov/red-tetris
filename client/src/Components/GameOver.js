import React from "react";
import url from "../img/Gameover.jpeg";
import url1 from "../img/Tetris-Founder.png";
import { StyledOverlay, StyledOverlayText } from "./styling/StyledStage";

export default function GameOver({ player }) {
  return (
    <StyledOverlay>
      <StyledOverlayText>
        {player?.lost === true ? (
          <img data-testid="gameOver-Lost-img" style={{ width: "255px" }} src={url} />
        ) : (
          <img data-testid="gameOver-Won-img" style={{ width: "255px" }} src={url1} />
        )}
        <h1>{player?.lost === true ? "You lost! Try HARD!" : "You Win! Congratulations! "}</h1>
        <p>
          {player?.admin === true
            ? "Restart the game"
            : "Wait for host to restart the game"}
        </p>
      </StyledOverlayText>
    </StyledOverlay>
  );
}
