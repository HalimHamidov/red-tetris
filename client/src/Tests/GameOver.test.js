import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import GameOver from "../Components/GameOver";
import React from 'react';

test("App render test ", () => {
  const tree = renderer.create(<GameOver />).toJSON();
  expect(tree).toMatchSnapshot();
});
test("Player Won ", () => {
  const { findByText, getByTestId } = render(
    <GameOver player={{ lost: false, admin: true }} />
  );
  expect(getByTestId("gameOver-Won-img")).toBeTruthy();
  const msg = "You Won";
  expect(findByText(msg)).toBeDefined();
});
