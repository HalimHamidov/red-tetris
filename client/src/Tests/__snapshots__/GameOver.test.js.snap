// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render test  1`] = `
<div
  className="sc-gsDKAQ hbqSUT"
>
  <div
    className="sc-dkPtRN gnDrrL"
  >
    <img
      data-testid="gameOver-Won-img"
      src="Tetris-Founder.png"
      style={
        Object {
          "width": "255px",
        }
      }
    />
    <h1>
      You Win! Congratulations! 
    </h1>
    <p>
      Wait for host to restart the game
    </p>
  </div>
</div>
`;
